src/app/lib/utilities/preprocessor/rules/timestamp.js

```diff
@@ -37,13 +37,26 @@ const insertTimestampBlock = (originalJson, timestampBlock) => {
   const { headlineBlocks, mainBlocks } = splitBlocksByHeadline(
     json.content.model.blocks,
   );
+
   if (headlineBlocks.length > 0) {
-    // insert timestamp block immediately after headline
-    json.content.model.blocks = [
-      ...headlineBlocks,
-      timestampBlock,
-      ...mainBlocks,
-    ];
+    const imageOrAresMediaFirst =
+      mainBlocks.length > 0 &&
+      (mainBlocks[0].type === 'image' || mainBlocks[0].type === 'aresMedia');
+
+    if (imageOrAresMediaFirst) {
```

Comment:
Code Climate is reporting that this addition is adding cognitive complexity (likely due to nested if statements) plus this is making this function longer than is recommended. 

To address the latter point, I'd suggest pulling out this `imageOrAresMediaFirst` to outside of the `insertTimestampBlock` function & making it a helper function that takes in an array of blocks. 
Then here `imageOrAresMediaFirst(mainBlocks)` can be used.